clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -2857,32 +2857,59 @@ static void handleWeakImportAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
 }
 
 // Checks correctness of mutual usage of different work_group_size attributes:
-// reqd_work_group_size, max_work_group_size. Values of reqd_work_group_size
-// arguments shall be equal or less than values coming from max_work_group_size.
+// reqd_work_group_size, max_work_group_size and max_global_work_dim.
+// Values of reqd_work_group_size arguments shall be equal or less than values
+// coming from max_work_group_size.
+// In case of 'max_global_work_dim' attribute equals to 0 we shall ensure
+// that if max_work_group_size and reqd_work_group_size attributes exist,
+// then they are equal (1, 1, 1).
 static bool checkWorkGroupSizeValues(Sema &S, Decl *D, const ParsedAttr &Attr,
                                      uint32_t WGSize[3]) {
-  if (const SYCLIntelMaxWorkGroupSizeAttr *A =
-      D->getAttr<SYCLIntelMaxWorkGroupSizeAttr>()) {
+  bool Result = true;
+  auto checkZeroDim = [](auto &S, auto &A, auto &Attr, auto X, auto Y,
+                         auto Z) -> bool{
+    if (X != 1 || Y != 1 || Z != 1) {
+      S.Diag(A->getLocation(), diag::err_sycl_x_y_z_arguments_must_be_one)
+          << A << Attr;
+      return false;
+    }
+    return true;
+  };
```

Comment:
This lambda does not capture anything at all. Moving it to a distinct method will improve readability here.

Reply:
I was thinking, that S - is a class field and therefore isn't being captured by lambda, because the caller is a static member. I was wrong.